cmd/collectors/ems/templating.go

```diff
@@ -0,0 +1,139 @@
+package ems
+
+import (
+	"github.com/netapp/harvest/v2/cmd/poller/collector"
+	"github.com/netapp/harvest/v2/cmd/poller/plugin"
+	"github.com/netapp/harvest/v2/pkg/tree/node"
+	"strings"
+)
+
+var defaultLabels = map[string]string{
+	"message.name":     "message",
+	"node.name":        "node",
+	"message.severity": "severity",
+	"time":             "time",
+	"index":            "index",
+}
+
+var defaultInstanceKey = []string{"index", "message.name"}
+
+func (e *Ems) ParseMatches(matches *node.Node, prop *emsProp) {
+	for _, v := range matches.GetChildren() {
+		name := v.GetChildContentS("name")
+		value := v.GetChildContentS("value")
+		if name != "" && value != "" {
+			prop.Matches = append(prop.Matches, &Matches{
+				Name:  name,
+				value: value,
+			})
+		} else {
+			e.Logger.Warn().Str("name", prop.Name).Msg("Missing matches for ems")
+			continue
+		}
+	}
+}
+
+func (e *Ems) ParseLabels(labels *node.Node, prop *emsProp) {
+	prop.Labels = make(map[string]string)
+	for _, v := range labels.GetChildren() {
+		labelName := v.GetNameS()
+		labelValue := v.GetContentS()
+		if labelName != "" && labelValue != "" {
+			prop.Labels[labelName] = labelValue
+		} else {
+			e.Logger.Warn().Str("name", prop.Name).Msg("Missing labels for ems")
+			continue
+		}
+	}
+}
+
+func (e *Ems) ParseRestCounters(counter *node.Node, prop *emsProp) {
+	var (
+		display, name, kind string
+	)
+
+	//load default ems labels
+	for k, v := range defaultLabels {
+		prop.InstanceLabels[k] = v
+	}
+
+	//load default instance keys
+	for _, v := range defaultInstanceKey {
+		prop.InstanceKeys = append(prop.InstanceKeys, v)
+	}
+
+	for _, c := range counter.GetAllChildContentS() {
+		if c != "" {
+			name, display, kind = ParseEmsMetric(c)
+			e.Logger.Debug().
+				Str("kind", kind).
+				Str("name", name).
+				Str("display", display).
+				Msg("Collected")
+
+			prop.Counters[name] = display
+			// EMS only supports labels
+			prop.InstanceLabels[name] = display
+		}
+	}
+	// add a placeholder metric for ems
+	m := &Metric{Label: "events", Name: "events", MetricType: "", Exportable: true}
+	prop.Metrics["events"] = m
+}
+
+func ParseEmsMetric(rawName string) (string, string, string) {
+	var (
+		name, display string
+		values        []string
+	)
+	if values = strings.SplitN(rawName, "=>", 2); len(values) == 2 {
+		name = strings.TrimSpace(values[0])
+		display = strings.TrimSpace(values[1])
+	} else {
+		name = rawName
+		display = strings.ReplaceAll(rawName, ".", "_")
+		display = strings.ReplaceAll(display, "-", "_")
+	}
+
+	return name, display, "label"
+}
+
+//LoadEmsPlugins loads built-in plugins or dynamically loads custom plugins
+//and adds them to the collector
+func (e *Ems) LoadEmsPlugins(params *node.Node) ([]plugin.Plugin, error) {
```

Comment:
almost same as collector class, can be re-used after modification in last few lines in collector.go